File: Code.java
// Definition for singly-linked list:
// class ListNode<T> {
//   ListNode(T x) {
//     value = x;
//   }
//   T value;
//   ListNode<T> next;
// }
//

ListNode<Integer> removeKFromList(ListNode<Integer> l, int k) {
    ListNode<Integer> removeKFromList = new ListNode<Integer>(0);
    removeKFromList.next = l;
    ListNode<Integer> temp_node = removeKFromList;
    
    while(temp_node.next != null)
    {
        if(temp_node.next.value == k)
            temp_node.next=temp_node.next.next;
        else
            temp_node=temp_node.next;
    }
    
    return removeKFromList.next;
}

File: Code.c
//
// Definition for singly-linked list:
// typedef struct list_##name {
//   type value;
//   struct list_##name *next;
// } list_##name;
//
// list_##name *alloc_list_##name(type v) {
//   list_##name *l = calloc(1, sizeof(list_##name));
//   l->value = v;
//   return l;
// }
//
list_integer * removeKFromList(list_integer * l, int k) {

        if(l != NULL){
                if(l->value == k)
                        return removeKFromList(l->next, k);
                else
                        l->next = removeKFromList(l->next, k);
        }
        
        return l;
}
